connaisseur/sigstore_validator.py

```diff
@@ -0,0 +1,78 @@
+import json
+import logging
+import re
+import subprocess  # nosec
+
+from connaisseur.crypto import decode_and_verify_ecdsa_key
+from connaisseur.exceptions import (
+    CosignError,
+    CosignTimeout,
+    NotFoundException,
+    ValidationError,
+)
+
+
+def get_cosign_validated_digests(image: str, pubkey: str):
+    """
+    Gets and processes cosign validation output for a given `image` and `pubkey`
+    and either returns a list of valid digests or raises a suitable exception
+    in case no valid signature is found or cosign fails.
+    """
+    failed, stdout, stderr = invoke_cosign(image, pubkey)
```

Comment:
`failed` is a misnomer, as it sounds boolean, but it is an int.

Reply:
renamed to `returncode` as in the subprocess module